bool armDisarm
int32 mode
